File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v1/SpotMarketApi.java
package com.bitget.openapi.api.v1;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface SpotMarketApi {

    @GET("/api/spot/v1/public/currencies")
    Call<ResponseResult> currencies();

    @GET("/api/spot/v1/public/products")
    Call<ResponseResult> products();

    @GET("/api/spot/v1/public/product")
    Call<ResponseResult> product(@QueryMap Map<String, String> paramMap);

    @GET("/api/spot/v1/market/fills")
    Call<ResponseResult> fills(@QueryMap Map<String, String> paramMap);

    @GET("/api/spot/v1/market/depth")
    Call<ResponseResult> depth(@QueryMap Map<String, String> paramMap);

    @GET("/api/spot/v1/market/ticker")
    Call<ResponseResult> ticker(@QueryMap Map<String, String> paramMap);

    @GET("/api/spot/v1/market/tickers")
    Call<ResponseResult> tickers();

    @GET("/api/spot/v1/market/candles")
    Call<ResponseResult> candles(@QueryMap Map<String, String> paramMap);
}

File: bitget-php-sdk-api/src/api/v1/SpotMarketApi.php
<?php


namespace bitget\api\v1;


use bitget\internal\BitgetApiClient;

class SpotMarketApi
{
    var BitgetApiClient $BitgetApiClient;

    public function __construct($BitgetApiClient)
    {
        $this->BitgetApiClient = $BitgetApiClient;
    }

    public function currencies($params): string
    {
        return $this->BitgetApiClient->doGet("/api/spot/v1/public/currencies", $params);
    }

    public function products($params): string
    {
        return $this->BitgetApiClient->doGet("/api/spot/v1/public/products", $params);
    }

    public function product($params): string
    {
        return $this->BitgetApiClient->doGet("/api/spot/v1/public/product", $params);
    }

    public function fills($params): string
    {
        return $this->BitgetApiClient->doGet("/api/spot/v1/market/fills", $params);
    }

    public function depth($params): string
    {
        return $this->BitgetApiClient->doGet("/api/spot/v1/market/depth", $params);
    }

    public function ticker($params): string
    {
        return $this->BitgetApiClient->doGet("/api/spot/v1/market/ticker", $params);
    }

    public function tickers($params): string
    {
        return $this->BitgetApiClient->doGet("/api/spot/v1/market/tickers", $params);
    }

    public function candles($params): string
    {
        return $this->BitgetApiClient->doGet("/api/spot/v1/market/candles", $params);
    }
}